airflow-core/src/airflow/models/trigger.py

```diff
@@ -192,6 +194,15 @@ def fetch_trigger_ids_with_asset(cls, session: Session = NEW_SESSION) -> set[str
         query = select(asset_trigger_association_table.columns.trigger_id)
         return {trigger_id for trigger_id in session.scalars(query)}
 
+    @classmethod
+    @provide_session
+    def fetch_trigger_ids_with_deadline_callback(cls, session: Session = NEW_SESSION) -> set[str]:
+        """Fetch all the trigger IDs associated with a Deadline Alert."""
+        from airflow.models.deadline import Deadline
+
+        query = select(Deadline.trigger_id).where(Deadline.trigger_id.is_not(None))
+        return {trigger_id for trigger_id in session.scalars(query)}
```

Comment:
Hmmmm, I'm not a fan of this pattern -- we now have it in two places, we might need to think about a more general purpose way than adding another special cased column/relationship